tests/Unit/Evolution/Systems/Ccz4/Test_Ricci.cpp

```diff
@@ -155,23 +162,49 @@ void test_compute_spatial_ricci_tensor(
       get<Tags::deriv<field_p_tag, tmpl::size_t<SpatialDim>, Frame::Inertial>>(
           d_field_p_var);
 
-  const auto christoffel_second_kind = Ccz4::conformal_christoffel_second_kind(
-      inverse_conformal_spatial_metric, field_d);
+  const auto conformal_christoffel_second_kind =
+      Ccz4::conformal_christoffel_second_kind(inverse_conformal_spatial_metric,
+                                              field_d);
+
+  const auto christoffel_second_kind = Ccz4::christoffel_second_kind(
+      conformal_spatial_metric, inverse_conformal_spatial_metric, field_p,
+      conformal_christoffel_second_kind);
+
+  using christoffel_second_kind_tag =
+      gr::Tags::SpatialChristoffelSecondKind<SpatialDim, Frame::Inertial,
+                                             DataVector>;
+  Variables<tmpl::list<christoffel_second_kind_tag>>
+      christoffel_second_kind_var(num_points_3d);
+  get<christoffel_second_kind_tag>(christoffel_second_kind_var) =
+      christoffel_second_kind;
+  const auto d_christoffel_second_kind_var =
+      partial_derivatives<tmpl::list<christoffel_second_kind_tag>>(
+          christoffel_second_kind_var, mesh, coord_map.inv_jacobian(x_logical));
+  const auto& d_christoffel_second_kind =
+      get<Tags::deriv<christoffel_second_kind_tag, tmpl::size_t<SpatialDim>,
+                      Frame::Inertial>>(d_christoffel_second_kind_var);
 
   // Compute expected and actual ricci tensors using above computed arguments
-  const auto expected_ricci_tensor{
+  const auto expected_python_ricci_tensor{
       pypp::call<tnsr::ii<DataVector, SpatialDim, Frame::Inertial>>(
           "Ricci", "spatial_ricci_tensor", christoffel_second_kind,
           d_conformal_christoffel_second_kind, conformal_spatial_metric,
           inverse_conformal_spatial_metric, field_d, field_d_up, field_p,
           d_field_p)};
 
+  const auto expected_cpp_gr_ricci_tensor =
+      gr::ricci_tensor(christoffel_second_kind, d_christoffel_second_kind);
+
   const auto actual_ricci_tensor = Ccz4::spatial_ricci_tensor(
       christoffel_second_kind, d_conformal_christoffel_second_kind,
       conformal_spatial_metric, inverse_conformal_spatial_metric, field_d,
       field_d_up, field_p, d_field_p);
 
-  CHECK_ITERABLE_APPROX(expected_ricci_tensor, actual_ricci_tensor);
+  CHECK_ITERABLE_APPROX(expected_python_ricci_tensor, actual_ricci_tensor);
+
+  Approx approx = Approx::custom().epsilon(1e-9).scale(1.0);
```

Comment:
not a change request, I'm just surprised the error is at 1e-9 rather than 1e-13 level. Do you understand why? This is sort of suspicious

Reply:
I don't know why. However, I wouldn't really say I have a baseline model in my head of how far off it should be given that numerical derivatives were used. I did check that intermediate tensors (approximately) agreed with expected, and I believe those required either 1e-10 or 1e-12 error. Would you like me to follow up on this computation with more tests? Or I can add back in the `CHECK`s for the intermediate values in this test. Let me know!

Reply:
I would print the l2norm (Blaze has support for this) of the difference `expected-computed` for grid size 2, 4, 6, 8, 10. Hopefully you see some sort of convergence :)

Reply:
Oh, is it not unusual for it to not monotonically decrease? The L2 norm seems to be converging on a minimum for `num_points_1d = 6`. I would have naively expected it to monotonically decrease as grid points increased just because I'm thinking more grid points == more accurate.

Reply:
Reporting back:

Dr. Lovelace noticed that my coordinates for this test were inside the horizon and suggested I make the coordinates farther away. Here are the parameters I originally used (on this PR):
```
  const double mass = 2.;
  const std::array<double, 3> spin{{0.3, 0.5, 0.2}};
  const std::array<double, 3> center{{0.2, 0.3, 0.4}};
  const size_t grid_size = 8;
  const std::array<double, 3> lower_bound{{0.82, 1.24, 1.32}};
  const std::array<double, 3> upper_bound{{0.8, 1.22, 1.3}};
```
And when I increase the bounds values by a factor 10 to:
```
  const std::array<double, 3> lower_bound{{8.2, 12.4, 13.2}};
  const std::array<double, 3> upper_bound{{8.0, 12.2, 13.0}};
```
the test passes with `1e-11` at best instead of `1e-9`. Note that, as before, increasing the number of grid points just makes the difference worse.

Do you have thoughts about this?

Reply:
Sounds like one of (or both) of the ways you're calculating things has bad cancellation errors. This will effectively mean that the CCZ4 system would ever be accurate beyond 1e-9, which isn't great. Keep in mind that with CCZ4 you're simulating inside the horizon so moving further out isn't a luxury you have in this case. I still find this worrying. I think the KS solution has an analytic expression for the spatial christoffel symbols that you could use instead of numerical derivatives to gain some insight into which expression is bad.